src/app-layer-parser.c

```diff
@@ -1219,11 +1234,10 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
 
     /* invoke the recursive parser, but only on data. We may get empty msgs on EOF */
     if (input_len > 0 || (flags & STREAM_EOF)) {
+        Setup(f, flags & (STREAM_TOSERVER | STREAM_TOCLIENT), input, input_len, &stream_slice);
```

Comment:
Q: So, if we are at the end of a stream, we might end up creating an empty stream slice here?

Reply:
Yes but that is fine. The parser should be accepting that as an EOF notification.